server/repository/src/db_diesel/assets/asset_log.rs

```diff
@@ -0,0 +1,205 @@
+        self
+    }
+}
+
+pub struct AssetLogRepository<'a> {
+    connection: &'a StorageConnection,
+}
+
+impl<'a> AssetLogRepository<'a> {
+    pub fn new(connection: &'a StorageConnection) -> Self {
+        AssetLogRepository { connection }
+    }
+
+    pub fn count(&self, filter: Option<AssetLogFilter>) -> Result<i64, RepositoryError> {
+        let query = create_filtered_query(filter);
+        Ok(query.count().get_result(&self.connection.connection)?)
+    }
+
+    pub fn query_one(&self, filter: AssetLogFilter) -> Result<Option<AssetLog>, RepositoryError> {
+        Ok(self.query_by_filter(filter)?.pop())
+    }
+
+    pub fn query_by_filter(
+        &self,
+        filter: AssetLogFilter,
+    ) -> Result<Vec<AssetLog>, RepositoryError> {
+        self.query(Pagination::all(), Some(filter), None)
+    }
+
+    pub fn query(
+        &self,
+        pagination: Pagination,
+        filter: Option<AssetLogFilter>,
+        sort: Option<AssetLogSort>,
+    ) -> Result<Vec<AssetLog>, RepositoryError> {
+        let mut query = create_filtered_query(filter);
+
+        if let Some(sort) = sort {
+            match sort.key {
+                AssetLogSortField::LogDatetime => {
+                    apply_sort!(query, sort, asset_log_dsl::log_datetime);
+                }
+                AssetLogSortField::Status => {
+                    apply_sort_no_case!(query, sort, asset_log_dsl::status);
+                }
+            }
+        } else {
+            query = query.order(asset_log_dsl::id.asc())
+        }
+
+        let final_query = query
+            .offset(pagination.offset as i64)
+            .limit(pagination.limit as i64);
+
+        // Debug diesel query
+        // println!(
+        //    "{}",
+        //     diesel::debug_query::<DBType, _>(&final_query).to_string()
+        // );
+
+        let result = final_query.load::<AssetLog>(&self.connection.connection)?;
+
+        Ok(result.into_iter().map(to_domain).collect())
+    }
+}
+
+fn to_domain(asset_log_row: AssetLogRow) -> AssetLog {
+    asset_log_row
+}
+
+type BoxedAssetLogQuery = IntoBoxed<'static, asset_log::table, DBType>;
+
+fn create_filtered_query(filter: Option<AssetLogFilter>) -> BoxedAssetLogQuery {
+    let mut query = asset_log_dsl::asset_log.into_boxed();
+
+    if let Some(f) = filter {
+        let AssetLogFilter {
+            id,
+            asset_id,
+            status,
+            log_datetime,
+        } = f;
+
+        apply_equal_filter!(query, id, asset_log_dsl::id);
+        apply_string_filter!(query, status, asset_log_dsl::status);
+        apply_date_filter!(query, log_datetime, asset_log_dsl::log_datetime);
+
+        if let Some(asset_id) = asset_id {
+            let mut sub_query = asset_dsl::asset.select(asset_dsl::id).into_boxed();
+            apply_equal_filter!(sub_query, Some(asset_id), asset_dsl::id);
+            query = query.filter(asset_log_dsl::asset_id.eq_any(sub_query));
+        }
```

Comment:
```suggestion
apply_equal_filter!(query, asset_id, asset_log_dsl::asset_id);
```
This doesn't need to be a subquery, because the `asset_id` field is already in the `asset_log` table, we need a subquery where we need to look up the id in another table, so for example if we wanted a filter to look up logs based on `Asset:Name` we'd need to do a subquery to look up the asset with that name, then put the `asset_id` we found into the filter.